compiler/passes/docs.cpp

```diff
@@ -58,73 +51,34 @@ static int compareClasses(const void *v1, const void* v2) {
-    if (fDocsTextOnly) {
-      outputMap["class"] = "Class: ";
-      outputMap["record"] = "Record: ";
-      outputMap["config"] = "config ";
-    } else {
-      outputMap["class"] = ".. class:: ";
-      outputMap["record"] = ".. record:: ";
-      outputMap["module"] = ".. module:: ";
-      outputMap["module comment prefix"] = ":synopsis: ";
-      outputMap["iter func"] = ".. iterfunction:: ";
-      outputMap["iter method"] = ".. itermethod:: ";
-      outputMap["func"] = ".. function:: ";
-      outputMap["method"] = ".. method:: ";
-      outputMap["config"] = ".. data:: config ";
-      outputMap["global"] = ".. data:: ";
-      outputMap["field"] = ".. attribute:: ";
-    }
-
     // Open the directory to store the docs
     std::string docsDir = (strlen(fDocsFolder) != 0) ? fDocsFolder : "docs";
-    std::string folderName = docsDir;
+    std::string docsFolderName = docsDir;
 
     if (!fDocsTextOnly) {
-      folderName = generateSphinxProject(docsDir);
+      docsFolderName = generateSphinxProject(docsDir);
     }
 
-    mkdir(folderName.c_str(), S_IWUSR|S_IRUSR|S_IXUSR);
+    mkdir(docsFolderName.c_str(), S_IWUSR|S_IRUSR|S_IXUSR);
+
     
     forv_Vec(ModuleSymbol, mod, gModuleSymbols) {
       // TODO: Add flag to compiler to turn on doc dev only output
       if (!mod->hasFlag(FLAG_NO_DOC) && !devOnlyModule(mod)) {
-        std::string filename = mod->filename;
-
-        if (mod->modTag == MOD_INTERNAL) {
-          filename = "internal-modules/";
-        } else if (mod ->modTag == MOD_STANDARD) {
-          filename = "standard-modules/";
-        } else {
-          size_t location = filename.rfind("/");
-          if (location != std::string::npos) {
-            filename = filename.substr(0, location + 1);
-          } else {
-            filename = "";
-          }
-        }
-        filename = folderName + "/" + filename;
-        createDocsFileFolders(filename);
-        
         if (isNotSubmodule(mod)) {
-          // Creates files for each top level module
-          if (!fDocsTextOnly)
-            filename = filename + mod->name + ".rst";
-          else
-            filename = filename + mod->name + ".txt";
-          std::ofstream file(filename.c_str(), std::ios::out);
-          if (!fDocsTextOnly) {
-            file << ".. default-domain:: chpl" << std::endl << std::endl;
-          }
-          printModule(&file, mod, mod->name);
-          file.close();
+          std::ofstream *file = openFileFromMod(mod, docsFolderName);
+
+          AstPrintDocs *docsVisitor = new AstPrintDocs(file);
+          mod->accept(docsVisitor);
+          delete docsVisitor;
+
+          // Comment the above three lines and uncomment the following line to
+          // get the old category based output (or alphabetical). Note that
+          // this will be restored (hopefully soon)... (thomasvandoren, 2015-02-22)
+          //
+          // printModule(file, mod, 0);
```

Comment:
While enabling it is only three lines, supporting multiple output formats adds a lot more complexity. I'm going to leave it out for now, since the goal is to document standard modules for the release with the source based ordering.